public/app/features/alerting/unified/hooks/useAlertManagerSourceName.ts

```diff
@@ -1,67 +1,17 @@
-import { useCallback } from 'react';
-
-import { useQueryParams } from 'app/core/hooks/useQueryParams';
-import store from 'app/core/store';
-
-import { ALERTMANAGER_NAME_LOCAL_STORAGE_KEY, ALERTMANAGER_NAME_QUERY_KEY } from '../utils/constants';
-import { AlertManagerDataSource, GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
-
-function useIsAlertManagerAvailable(availableAlertManagers: AlertManagerDataSource[]) {
-  return useCallback(
-    (alertManagerName: string) => {
-      const availableAlertManagersNames = availableAlertManagers.map((am) => am.name);
-      return availableAlertManagersNames.includes(alertManagerName);
-    },
-    [availableAlertManagers]
-  );
-}
+import { useSelectedAlertmanager } from '../state/AlertmanagerContext';
 
 /* This will return am name either from query params or from local storage or a default (grafana).
  * Due to RBAC permissions Grafana Managed Alert manager or external alert managers may not be available
  * In the worst case neither GMA nor external alert manager is available
  */
+// TODO remove this fn and use useSelectedAlertmanager(); everywhere
+interface Props {
+  withPermissions: 'instance' | 'notification';
```

Comment:
I don't like this, ideally the scope of the permissions should be passed in to the `AlertPageContextProvider`